### PR TITLE
docs for debugging + debug improvements

### PR DESCRIPTION
**Pull request by grosser**
@zendesk/compute

### CODE REVIEW

File: kubernetes/debug.yml [hunk 1]
```diff
@@ -47,6 +47,8 @@ spec:
         cpu: '0.5'
         memory: '300Mi'
     command: ["sleep", "9999"]
```
Comment (uthark) [inline kubernetes/debug.yml]: `cat -v` also works and doesn't have a timeout.

File: kubernetes/debug.yml [hunk 2]
```diff
@@ -47,6 +47,8 @@ spec:
...
+    securityContext:
+      runAsUser: 0
```
Comment (uthark) [inline kubernetes/debug.yml]: Why does it need root?

File: cloudbuild.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+steps:
+  - name: 'gcr.io/cloud-builders/docker'
+    args:
+      - 'build'
+      - '--tag=gcr.io/$PROJECT_ID/apps/samson_secret_puller:$SHORT_SHA'
```
Comment (uthark) [inline cloudbuild.yaml]: Using git tags is more descriptive.

Comment (grosser) [reply cloudbuild.yaml]: yes, but then we need fuzzy tags too (when commit is not tagged) which will get ugly
also most other projects use revision, so consistency is nice too :)

File: kubernetes/debug.yml (hunk 2 above)
Comment (grosser) [reply kubernetes/debug.yml]: so you can install/change while debugging

File: kubernetes/debug.yml (hunk 1 above)
Comment (grosser) [reply kubernetes/debug.yml]: I'd rather have the timeout so it eventually dies ... also `sleep` is more readable

**Pull request merged**